### PR TITLE
autokey cipher added in c, c++ and java

### PR DESCRIPTION
**Pull request by ravireddy07**
**Fixes issue:**
 #4443 

**Changes:**

-- added Auto Key Cipher in C, C++ and Java.

### CODE REVIEW

File: code/cryptography/src/autokey_cipher/autokey.c [hunk 1]
```diff
@@ -0,0 +1,87 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+void flush_stdin()
+{
+    while(fgetc(stdin) != '\n');
+}
+
+void AutoKeyCipher(char** Text, char key, int encrypt)
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.c]: lowerCamelCase for all names/params.

File: code/cryptography/src/autokey_cipher/autokey.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+{
+    int textSize, i, nextKey, keyValue, result;
+
+    textSize = strlen((*Text));
+
+    for(i = 0; i < textSize; i++)
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.c]: Pre increment in all loops.

File: code/cryptography/src/autokey_cipher/autokey.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#include <string.h>
+
+void flush_stdin()
+{
+    while(fgetc(stdin) != '\n');
+}
+
+void AutoKeyCipher(char** Text, char key, int encrypt)
+{
+    int textSize, i, nextKey, keyValue, result;
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.c]: Declare these near where they are first used; no need to declare everything at the top of the function.

File: code/cryptography/src/autokey_cipher/autokey.c [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    {
+        if(isalpha((*Text)[i]))
+        {
+            (*Text)[i] = toupper((*Text)[i]);
+        }
+    }
+
+    nextKey = toupper(key) - 'A';
+
+    for(i = 0; i < textSize; i++)
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.c]: Reformat this file using clang-format with 4 spaces.

File: code/cryptography/src/autokey_cipher/autokey.cpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+#include <stdlib.h>
+#include <string.h>
+using namespace std;
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.cpp]: Don't use this.

File: code/cryptography/src/autokey_cipher/autokey.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+#include <stdlib.h>
+#include <string.h>
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.cpp]: Just use regular std::string.

File: code/cryptography/src/autokey_cipher/autokey.cpp [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+using namespace std;
+
+void flush_stdin()
+{
+    while(fgetc(stdin) != '\n');
+}
+
+void AutoKeyCipher(char** Text, char key, int encrypt)
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.cpp]: Same naming related things. Also use std::string.

File: code/cryptography/src/autokey_cipher/autokey.cpp [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+{
+    int textSize, i, nextKey, keyValue, result;
+    textSize = strlen((*Text));
+    for(i=0;i<textSize;i++)
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.cpp]: In C++, the loop variable can be declared in the loop directly.

File: code/cryptography/src/autokey_cipher/autokey.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+
+public class autokey
+{
+	public static String createKey(String plainText, int keyCount, HashMap<Character, Integer> hashMap1, HashMap<Integer, Character> hashMap2)
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.java]: Reformat this file to use 4 space indentation.

File: code/cryptography/src/autokey_cipher/autokey.c [hunk 10]
```diff
@@ -0,0 +1,87 @@
+#include <stdio.h>
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey.c]: Also please rename all files in this directory to `autokey_cipher`.

File: code/cryptography/src/autokey_cipher/autokey_cipher.cpp [hunk 11]
```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include <string.h>
+
+void flush_stdin()
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey_cipher.cpp]: This function isn't used in this file, so just remove it. Also include the `<string.h>` header.

File: code/cryptography/src/autokey_cipher/autokey_cipher.cpp [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+{
+    while (fgetc(stdin) != '\n');
+}
+
+std::string autokeycipher(std::string text, char key, int encrypt)
+{
+    int textsize;
+    textsize = sizeof(text) / sizeof(text[0]);
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey_cipher.cpp]: This is incorrect. Use `string.length()` and don't store it in a variable. Also, you need to pass the string as a reference, or else it won't be modified.

**Comment by ravireddy07** [comment]
@arnavb  Requested changes made. Please check now.

**Comment by arnavb** [comment]
@ravireddy07 I think there is an issue with this PR because lots of unrelated files are being pushed. Try merging the latest code from master in. If that doesn't fix (and anything else you tried doesn't work either), then try recreating this PR.

**Comment by ravireddy07** [comment]
@arnavb  Due to case-sensitive file names or folder names, my local master/branch is replacing those files. Due to that, all unwanted files getting commited along with my changes in pr. Now, i've created another PR: #4788 . Please review that once.

i'm working on this since last 2weeks. due to this conflicts or unwanted case-sensitive files, i'm not able to contribute on other issues. I've made the code on 2 other issues. but, due to this errors, not created pr yet.

@arnavb  Please review this PR: #4788  and request if any changes required. I'll close this PR.

**Comment by ravireddy07** [comment]
Now, i enabled case-sensitive in all my directories. Things looks fine now..!!

**Pull request closed without merging**